File: backend/src/main/java/com/zufang/entity/House.java
package com.zufang.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 房屋实体类
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("houses")
public class House {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("house_name")
    private String houseName;
    
    @TableField("address")
    private String address;
    
    @TableField("total_rooms")
    private Integer totalRooms;
    
    @TableField("total_area")
    private BigDecimal totalArea;
    
    @TableField("house_type")
    private String houseType;
    
    @TableField("description")
    private String description;
    
    @TableField("facilities")
    private String[] facilities;
    
    @TableField("images")
    private String[] images;
    
    @TableField("landlord_id")
    private Long landlordId;
    
    @TableField("status")
    private String status;
    
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/house.ts
import request from './request';

export interface House {
  id?: number;
  houseName: string;
  address: string;
  totalRooms: number;
  totalArea: number;
  houseType: string;
  description: string;
  facilities: string;
  images: string;
  landlordId: number;
  status: string;
}

export interface HouseQuery {
  current: number;
  size: number;
  keyword?: string;
}

// 分页获取房屋列表
export function getHouseList(params: HouseQuery) {
  return request({
    url: '/houses/page',
    method: 'get',
    params
  });
}

// 获取房屋详情
export function getHouseDetail(id: number) {
  return request({
    url: `/houses/${id}`,
    method: 'get'
  });
}

// 添加房屋
export function addHouse(data: House) {
  return request({
    url: '/houses',
    method: 'post',
    data
  });
}

// 更新房屋
export function updateHouse(data: House) {
  return request({
    url: '/houses',
    method: 'put',
    data
  });
}

// 删除房屋
export function deleteHouse(id: number) {
  return request({
    url: `/houses/${id}`,
    method: 'delete'
  });
}

// 更新房屋状态
export function updateHouseStatus(id: number, status: string) {
  return request({
    url: `/houses/${id}/status/${status}`,
    method: 'put'
  });
}